src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java

```diff
@@ -81,10 +81,17 @@ public void init(ChunkSectionPos pos) {
                             this.tileEntities.put(ChunkSectionPos.packLocal(new BlockPos(tileentity.xCoord & 15, tileentity.yCoord & 15, tileentity.zCoord & 15)), tileentity);
                         }
                     }
-                    this.biomeData[(lZ << 4) | lX] = world.getBiomeGenForCoords(x, z);
                 }
             }
         }
+
+        // Fill biome data
+        for(int z = pos.getMinZ(); z <= pos.getMaxZ(); z++) {
+            for(int x = pos.getMinX(); x <= pos.getMaxX(); x++) {
+                int lZ = z & 15;
+                this.biomeData[((z & 15) << 4) | x & 15] = world.getBiomeGenForCoords(x, z);
```

Comment:
Please add brackets around (x & 15) and remove the redundant `lZ` variable, otherwise looks good.